M2/Macaulay2/m2/packages.m2

```diff
@@ -521,7 +516,15 @@ endPackage String := title -> (
      pkg.PackageIsLoaded = true;
      pkg)
 
-beginDocumentation = () -> (
+beginDocumentation = {
+    CacheExampleOutput        => null,
+    OptionalComponentsPresent => null,
+    UseCachedExampleOutput    => null, -- TODO: not used by any package, retire this!
```

Comment:
I use this option in the Debian package to fix some reproducibility issues and to cache some examples that occasionally crash, so I'd prefer to keep it around.

Reply:
`UseCachedExampleOutput`? Can you say more about how you use it? Maybe I don't understand it well. If cached examples are present, wouldn't M2 automatically use them? (I think it does, e.g. see the single `realrank` example that I added)

Reply:
I think it needs to be true in order to copy any cached examples over:

https://github.com/Macaulay2/M2/blob/ec9e9ac60ed4a8e791448942202f077a88a87e15/M2/Macaulay2/m2/installPackage.m2#L556-L562

I use it in the Debian package by [patching several packages](https://salsa.debian.org/math-team/macaulay2/-/blob/debian/latest/debian/patches/use-cached-examples.patch) so that `OptionalComponentsPresent` is true, and then copying the cached examples from [this directory](https://salsa.debian.org/math-team/macaulay2/-/tree/debian/latest/debian/examples) over to the corresponding package directories before building the docs.